src/assistant_library_demo.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A demo of the Google Assistant Library"""
+
+import sys
+import threading
+
+import aiy.assistant.auth_helpers
+import aiy.voicehat
+from google.assistant.library import Assistant
+from google.assistant.library.event import EventType
+
+
+class MyAssistant(object):
```

Comment:
Unfortunately a background thread is needed, the reason being that the assistant library completely blocks the main thread and prevents the GPIO driver to "interrupt" and call the _on_button_press() callback. So this is necessary to support the button trigger.

I am going to just revert this change and merge the pull request. This change has been pending for way too long, with the not-so-efficient North America - Europe collaboration...